pkg/controllers/user/pipeline/controller/pipelineexecution/pipelineexecution.go

```diff
@@ -129,3 +147,40 @@ func (l *Lifecycle) initLogs(obj *v3.PipelineExecution) error {
 	}
 	return nil
 }
+
+func (l *Lifecycle) getHeadCommit(execution *v3.PipelineExecution) (string, error) {
+	pipeline := execution.Spec.Pipeline
+	if !utils.ValidSourceCodeConfig(pipeline.Spec) {
+		return "", errors.New("invalid pipeline definition")
```

Comment:
Should we elaborate on pipeline specifics? So user can figure out which one is invalid. May be specify the id?